portfolio/src/main/webapp/task.js

```diff
@@ -6,20 +6,123 @@
  * converted into a JSON string.
  */
 class Task {
-  // TODO(raulcruise): could you add the type for each param please? 
-  
+  // TODO(raulcruise): could you add the type for each param please?
+
   /**
    * All fields are required except description, which will be null
    * when not set.
    * @param name: name of task as String
    * @param description: description of task as String, can be null
-   * @param duration: duration of task as Duration class
-   * @param priority: priority level of task, range 1-5 inclusive. 
+   * @param duration: duration of task as String class
+   * @param priority: priority level of task, range 1-5 inclusive.
    */
-  constructor(name, description, duration, priority) {
+  constructor(name, description, duration, taskPriority) {
     this.name = name;
     this.description = description;
     this.duration = duration;
-    this.priority = priority;
+    this.taskPriority = taskPriority;
   }
 }
+
+function createNewTask() {
```

Comment:
Can we have some comment? For example, which fields are required and which are optional? What does this method do?

Reply:
Done.

Reply:
Can you specify the return type of this function? It seems like in some code paths it returns `false` and in others it doesn't return anything at all. I suggest we tighten it up so that it either always returns a value or never does, and then if it is returning something you should define what it means.

Reply:
Done.